### PR TITLE
[WebKit Checkers] Allow a guardian CheckedPtr/CheckedRef

### PR DESCRIPTION
**Pull request by rniwa**
This PR makes WebKit checkers allow a guardian variable which is CheckedPtr or CheckedRef as in addition to RefPtr or Ref.

### CODE REVIEW

File: clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp [hunk 1]
```diff
@@ -56,7 +60,8 @@ bool tryToFindPtrOrigin(
       if (StopAtFirstRefCountedObj) {
         if (auto *ConversionFunc =
                 dyn_cast_or_null<FunctionDecl>(cast->getConversionFunction())) {
-          if (isCtorOfRefCounted(ConversionFunc))
+          if (isCtorOfRefCounted(ConversionFunc) ||
+              isCtorOfCheckedPtr(ConversionFunc))
```
Comment (t-rasmud) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp:64]: Could this also be refactored into a function (say `isSafeCtor`) like `isSafePtr`?

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h [hunk 2]
```diff
@@ -63,18 +63,29 @@ std::optional<bool> isUncounted(const clang::CXXRecordDecl* Class);
...
+bool isCtorOfCheckedPtr(const clang::FunctionDecl *F);
+
+/// \returns true if \p Name is RefPtr, Ref, or its variant, false if not.
+bool isRefType(const std::string &Name);
+
+/// \returns true if \p Name is CheckedRef or CheckedPtr, false if not.
+bool isCheckedPtr(const std::string &Name);
+
 /// \returns true if \p T is RefPtr, Ref, or its variant, false if not.
 bool isRefType(const clang::QualType T);
```
Comment (t-rasmud) [inline clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h:85]: This is more of a question for me to understand: Is there a reason this overload of `isRefType` isn't modified to check for `CheckedPtr`, `CheckedRef`?

File: clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp (hunk 1 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/ASTUtils.cpp:64]: Sure, we can do that.

File: clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h (hunk 2 above)
Comment (rniwa) [reply clang/lib/StaticAnalyzer/Checkers/WebKit/PtrTypesSemantics.h:85]: Apparently this was a dead code. Just deleted this declaration.

**Review by t-rasmud — APPROVED** [review]
LGTM!

**Pull request merged**